### PR TITLE
Update to Cats Effect v3.5.0

### PR DESCRIPTION
**Pull request by armanbilge**
Closes https://github.com/tpolecat/doobie/issues/1852.

### CODE REVIEW

**Comment by amumurst** [comment]
I went ahead and did the manual implementation of fromFutureCancelable in all WeakAsync implementations in https://github.com/armanbilge/doobie/pull/1/files. I don't really know much of this pattern, so just followed how fromFuture was implemented.

File: modules/free/src/main/scala/doobie/free/kleisliinterpreter.scala [hunk 1]
```diff
@@ -125,6 +125,9 @@ class KleisliInterpreter[M[_]](logHandler: LogHandlerM[M])(implicit val asyncM:
   def fromFuture[G[_], J, A](interpreter: G ~> Kleisli[M, J, *])(fut: Free[G, Future[A]]): Kleisli[M, J, A] = Kleisli(j =>
     asyncM.fromFuture(fut.foldMap(interpreter).run(j))
   )
+  def fromFutureCancelable[G[_], J, A](interpreter: G ~> Kleisli[M, J, *])(fut: Free[G, (Future[A], Free[G, Unit])]): Kleisli[M, J, A] = Kleisli(j =>
```
Comment (amumurst) [inline modules/free/src/main/scala/doobie/free/kleisliinterpreter.scala:128]: For a reviewer: As far as I know this (and the same one in the postgres module) is the only "interesting" details outside WeakAsync/WeakAsyncSuite (which is just based on fromFuture).

**Comment by Daenyth** [comment]
@armanbilge @amumurst it looks like there's some merge conflicts - once those are fixed this should be mergeable I think

**Comment by Daenyth** [comment]
Ahhh I misread github's UI - 
Screenshot 2023-05-17 at 9 07 24 AM

I thought it meant merge conflicts! Nevermind :)

**Review by Daenyth — APPROVED** [review]

**Comment by guymers** [comment]
The `freeGen2` command generates the files in `doobie.free.*` and `doobie.postgres.free.*`

**Comment by armanbilge** [comment]
@guymers thanks, pro-tip! Is the generator still actively used? If you just run it on `main` it makes a diff 😅 see https://github.com/tpolecat/doobie/commit/1ad22dbd3eec125a596b770ddb1691d15bba13c1

**Comment by armanbilge** [comment]
Ok, well in any case, I updated the generator to reflect the changes we made here, but I didn't actually run it.

**Comment by jatcwang** [comment]
That's fine @armanbilge we can raise another PR to rectify that. Thanks :)

**Comment by guymers** [comment]
> @guymers thanks, pro-tip! Is the generator still actively used? If you just run it on `main` it makes a diff sweat_smile see [1ad22db](https://github.com/tpolecat/doobie/commit/1ad22dbd3eec125a596b770ddb1691d15bba13c1)

It seems to get forgotten about but it's better than doing it manually ;). The methods generated depend on what version of Java was used, I assume the current files were generated using Java 8. It also generates some unused imports that were many removed to remove the warnings.

**Comment by armanbilge** [comment]
> The methods generated depend on what version of Java was used, I assume the current files were generated using Java 8.

Ahh, maybe that's the issue! I was using Java 17 locally. I'll give it one more try ... thanks!!

**Comment by ahjohannessen** [comment]
Any ETA on this one and RC3? :)

File: modules/free/src/main/scala/doobie/WeakAsync.scala [hunk 2]
```diff
@@ -7,13 +7,14 @@ package doobie
...
 import cats.effect.kernel.{ Async, Poll, Resource, Sync }
-import cats.effect.implicits._
 import cats.effect.std.Dispatcher
 import scala.concurrent.Future
 import scala.concurrent.duration.FiniteDuration
 
 trait WeakAsync[F[_]] extends Sync[F] {
   def fromFuture[A](fut: F[Future[A]]): F[A]
+  def fromFutureCancelable[A](fut: F[(Future[A], F[Unit])]): F[A] =
+    fromFuture(map(fut)(_._1)) // default impl for backwards compat; implementors should override
```
Comment (jatcwang) [inline modules/free/src/main/scala/doobie/WeakAsync.scala:17]: Happy to remove this default impl, as we are already breaking compatibility, so it's better to leave this out to ensure current/future implementations are handling cancellations :)

Comment (armanbilge) [inline modules/free/src/main/scala/doobie/WeakAsync.scala:17]:
```suggestion
  def fromFutureCancelable[A](fut: F[(Future[A], F[Unit])]): F[A]
```

**Comment by jatcwang** [comment]
:clinking_glasses:

**Comment by armanbilge** [comment]
Until the next RC, this fix is available in `1.0-f718b2e-SNAPSHOT`.
https://s01.oss.sonatype.org/content/repositories/snapshots/org/tpolecat/doobie-core_2.13/1.0-f718b2e-SNAPSHOT/

**Comment by ahjohannessen** [comment]
@armanbilge Was doobie-hikari also published?

**Comment by armanbilge** [comment]
Yes, everything was published.
https://s01.oss.sonatype.org/content/repositories/snapshots/org/tpolecat/doobie-hikari_2.13/1.0-f718b2e-SNAPSHOT/

**Pull request merged**